### DISCUSSION TITLE
WARNING: procps-ng 4.0.3 breaks lxqt-session — recompilation is required

### DISCUSSION THREAD

**Discussion by tsujan** [lxqt/lxqt#2443] [General]
I upgraded my system last night, and I couldn't log into LXQt from SDDM today.

Fortunately, `~/.local/share/sddm/xorg-session.log` told the reason: "lxqt-session: error while loading shared libraries: libprocps.so.8: cannot open shared object file: No such file or directory". The upgrade of `procps-ng`in Arch/Manjaro to 4.0.3-1 was the cause, and a simple recompilation of `lxqt-session` solved the problem.

I also noticed a problem in the Qt6-based pcmanfm-qt when I was forced to log into Wayland sessions (Wayfire and Plasma-Wayland) to recompile `lxqt-session`: Not only the "Create New" menu didn't work under Wayland anymore, but also it made the window of pcmanfm-qt totally unresponsive. Closing and opening of the window solved the unresponsiveness but "Create New" still didn't work. Recompilation didn't have any effect either. So, this may not be related to `procps-ng`, but I'm sure it didn't exist before...

EDIT: The cause of pcmanfm-qt's issue can be https://github.com/lxqt/libfm-qt/commit/f4ea04ca9333598c795a8a60fb68db1f0a5f788a. For some reason, it works with Qt5 but not with Qt6. I'll investigate it.

**Comment by stefonarch** [comment]
On arch it is coming in too today, didn't update for some days. but will wait until I've time to fiddle.

**Reply by tsujan** [reply]
Fixing `lxqt-session` didn't take my time today. But I hate it when Qt6 doesn't work as expected; I should fix that "Create New" dialog under Wayland....

**Comment by tsujan** [comment]
Ah, at last! The problem of "Create New" dialog in the Qt6-based pcmanfm-qt under Wayland is actually a regression in Qt6: `QWidget::minimumHeight()` is zero for the dialog before it's shown. It also happens with the bulk renaming dialog.

I found an elementary workaround, which is safer with Qt5 too; will make 2 PRs.

**Reply by stefonarch** [reply]
It was released  26.6 in arch, didn't see any topic in the forum. So  arch package needs  an update, making it depend on the version 4.03 @yan12125 ?

**Reply by tsujan** [reply]
I don't know about the Arch package, using self-compiled packages. But I think @yan12125 has taken care of it.

**Reply by stefonarch** [reply]
Updated and recompiled `lxqt-session` before closing session out (with sddm which did hang, not eve a tty, had to force shutdown) - all good.

**Reply by tsujan** [reply]
> with sddm which did hang

Yes, I saw that hang too and used the power button, which causes a normal shut-down here. No hang in SDDM after that. It was upgraded to 0.20.0-1 and was my first suspicion, but it was OK.

**Reply by yan12125** [reply]
Arch has SOP for such changes :)

https://archlinux.org/todo/libprocps-procps-ng-4x-rebuild/

**Reply by tsujan** [reply]
> Arch has SOP for such changes :)

Yes, I remember it from another case.

The warning was for users who compile LXQt. It was a bad experience seeing everything stopped before starting. It was also mixed with a totally unrelated issue.

**Comment by tsujan** [comment]
I applied the changes to the Qt6 branches of libfm-qt and pcmanfm-qt. As for libfm-qt, since I hadn't applied https://github.com/lxqt/libfm-qt/commit/f4ea04ca9333598c795a8a60fb68db1f0a5f788a before, no other user would see any problem under Wayland.
